### ISSUE TITLE
Screen Reader: Workspaces: Checkboxes are announced without part of the displayed text

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#76911]
## Action Performed:
Prerequisites:
- the user is signed in

1. Using Chrome + JAWS, open the URL https://new.expensify.com/
2. Navigate using the Tab key to the Workspaces section.
3. Move focus to the More options icon (three vertical dots) > Duplicate Workspace
4. Navigate using the tab key to the Next button and press Enter to activate it.
5. Navigate using the tab key to the checkboxes and observe the screen reader announcement.

## Expected Result:
Screen readers should announce both the main label and the associated descriptive text, providing full context, e.g.:
- 'Reports, 3 report fields'

## Actual Result:
When navigating to the Reports checkbox, the screen reader announces only the main label (“Eg: Reports”) and does not announce the additional descriptive text (“3 report fields”). As a result, users are not provided with complete context or details for what the checkbox represents.

HTML:
<div aria-label="Reports" role="presentation" data-tag="pressable" data-selection-scrapper-hidden-element="true" data-inner-box-shadow-element="false" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21" id="reports" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx" style=""><div class="css-g5y9jx r-13awgt0 r-1wtj0ep r-1awozwy r-18u37iz r-b5h31w r-1ah4tor r-9ffhgg r-lrvibr r-adacv r-kzbkwu r-ttdzmv" data-testid="base-list-item-reports" style=""><div class="css-g5y9jx r-13awgt0 r-1habvwh"><div dir="auto" class="css-146c3p1 r-dnmrzs r-1udbk01 r-1iln25a" style="color: rgb(231, 236, 233); font-size: 15px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 700; min-height: 20px; line-height: 20px; white-space: pre; text-decoration-line: none; overflow: hidden; margin-bottom: 4px;">Reports</div><div dir="auto" class="css-146c3p1 r-8akbws r-krxsd3 r-1qsk4np r-1udbk01" style="-webkit-line-clamp: 2; color: rgb(175, 187, 176); font-size: 13px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 400; line-height: 16px; white-space: pre-wrap; max-width: 303px;">3 report fields</div></div><div class="css-g5y9jx" data-component="AnimatedComponent" data-element="Component" data-source-file="createAnimatedComponent.js" style="opacity: 1;"><div aria-checked="true" aria-label="Reports" role="checkbox" data-tag="pressable" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx" style="justify-content: center; align-items: center; border-radius: 6px;"><div class="css-g5y9jx r-18og6j9 r-8drixv" style="height: 20px; width: 20px; border-width: 2px; justify-content: center; align-items: center; border-radius: 4px; margin: 2px;"><div class="css-g5y9jx" data-testid=""><svg xmlns="[http://www.w3.org/2000/svg"](http://www.w3.org/2000/svg%22) xml:space="preserve" viewBox="0 0 20 20" width="14" height="14" hovered="false" pressed="false" fill="#E7ECE9"><path d="M17.2 3.4c.9.7 1.1 1.9.4 2.8L8.9 18l-6.3-5.7c-.8-.7-.9-2-.1-2.8.7-.8 2-.9 2.8-.1l3 2.7 6.1-8.2c.6-1 1.9-1.2 2.8-.5" style="fill-rule: evenodd; clip-rule: evenodd;"></path></svg></div></div></div></div></div></div>

## Area issue was found in:
Workspace

## Failed WCAG checkpoints
WCAG 2.4.6 Headings and Labels
https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels

## User impact:
Descriptive headings are necessary for users who have disabilities that make reading slow and for people with limited short-term memory. These users benefit when section titles make it possible to predict what each section contains. Form input controls with labels that clearly describe the content that is expected to be entered helps users know how to successfully complete the form.

## Suggested resolution:
Ensure that both texts are either included in the current aria-label or associated with the checkbox by using <label> and <for .. ID>.
https://www.w3.org/WAI/tutorials/forms/labels/

## Environment(s)
Windows Windows 11 - Chrome

## Screenshots/Videos
https://github.com/user-attachments/assets/a4984567-86ab-4632-926b-362507844ef0

<img width="1919" height="967" alt="Image" src="https://github.com/user-attachments/assets/3e86cae0-cb4f-437b-857c-30b48f2f9af8" />

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6996772
**Issue reported by:** NA
**Slack conversation:** NA

**Comment by mavrickdeveloper** [comment]
### PROPOSAL

### Please re-state the problem that we are trying to solve in this issue.

When navigating to the Reports checkbox, the screen reader announces only the main label (“Eg: Reports”) and does not announce the additional descriptive text (“3 report fields”). As a result, users are not provided with complete context or details for what the checkbox represents.

### What is the root cause of that problem?

he issue cause is from two locations where accessibility labels are constructed using only `item.text`, completely ignoring the `item.alternateText `property:

https://github.com/Expensify/App/blob/d34df44329db681f7bdd1a8114cc44c41df161fe/src/components/SelectionList/ListItem/BaseListItem.tsx#L106

The outer pressable wrapper only uses `item.text `for its accessibility label. The `alternateText `(which contains the "3 report fields") is rendered visually but not included in the accessibility announcement

Next,MultiSelectListItem.tsx (Checkbox Comp):

https://github.com/Expensify/App/blob/d34df44329db681f7bdd1a8114cc44c41df161fe/src/components/SelectionList/ListItem/MultiSelectListItem.tsx#L34

Observation:The checkbox component within `MultiSelectListItem` also only uses `item.text`. This is what screen readers announce when users Tab to the checkbox

**Data Flow**

The data originates from `WorkspaceDuplicateSelectFeaturesForm.tsx`:

https://github.com/Expensify/App/blob/d34df44329db681f7bdd1a8114cc44c41df161fe/src/pages/workspace/duplicate/WorkspaceDuplicateSelectFeaturesForm.tsx#L186-L194

`text`: Main label (for eg"Reports")

`alternateText`: Descriptive detail (for eg: "3 report fields")

Both values are **correctly populated and passed,** but only text is used for accessibility

### What changes do you think we should make in order to solve the problem?

First, in `BaseListItem.tsx`:

https://github.com/Expensify/App/blob/d34df44329db681f7bdd1a8114cc44c41df161fe/src/components/SelectionList/ListItem/BaseListItem.tsx#L106

Should be updated to :

```tsx
accessibilityLabel={item.alternateText ? `${item.text}, ${item.alternateText}` : (item.text ?? '')}
```

This will fixes the issue for all `SelectionList` usages across the app
